src/Spectre.libClassifier/OpenCvDataset.h

```diff
@@ -0,0 +1,92 @@
+﻿#pragma once
+#include <span.h>
+#include <opencv2/core/mat.hpp>
+#include "Spectre.libDataset/IReadOnlyDataset.h"
+#include "Empty.h"
+
+namespace Spectre::libClassifier {
+	using DataType = float;
+	using Observation = gsl::span<const DataType>;
+	using Label = signed;
+	const auto CV_TYPE = CV_32FC1;
+	const auto CV_LABEL_TYPE = CV_32SC1;
+
+/// <summary>
+/// Data-owning structure which couples our typed with OpenCV cv::Mat
+/// </summary>
+class OpenCvDataset final : public Spectre::libDataset::IReadOnlyDataset<Observation, Label, Empty>
+{
+public:
+    /// <summary>
+    /// Initializes a new instance of the <see cref="OpenCvDataset"/> class.
+    /// </summary>
+    /// <param name="data">The data.</param>
+    /// <param name="labels">The labels.</param>
+    OpenCvDataset(gsl::span<const DataType> data, gsl::span<const Label> labels);
+    /// <summary>
+    /// Initializes a new instance of the <see cref="OpenCvDataset"/> class.
+    /// </summary>
+    /// <param name="data">The data.</param>
+    /// <param name="labels">The labels.</param>
+    OpenCvDataset(cv::Mat data, cv::Mat labels);
+
+	/// <summary>
+	/// Gets sample under specified index in read-only fashion.
+	/// </summary>
+	/// <param name="idx">The index.</param>
+	/// <returns>Sample</returns>
+	const Observation& operator[](size_t idx) const override;
+	/// <summary>
+	/// Gets the sample metadata in read-only fashion.
+	/// </summary>
+	/// <param name="idx">The index.</param>
+	/// <returns>Sample metadata</returns>
+	const Label& GetSampleMetadata(size_t idx) const override;
+	/// <summary>
+	/// Gets the dataset metadata in read-only fashion.
+	/// </summary>
+	/// <returns>Dataset metadata</returns>
+	const Empty& GetDatasetMetadata() const override;
```

Comment:
No. `Empty` is a `NullObject` pattern implementation. It is used now to avoid `void` in template parameters for `IReadOnlyDataset`, since `void` is not yet allowed to be passed (compiler cries).